packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -107,43 +118,97 @@ contract BridgePoolFactory is BridgePoolFactoryInterface, Ownable, OVM_CrossDoma
+     */
+    function setMinimumBridgingDelay(uint64 _minimumBridgingDelay, uint32 _l2Gas) public onlyOwner depositContractSet {
+        sendCrossDomainMessage(
+            depositContract,
+            _l2Gas,
+            abi.encodeWithSignature("setMinimumBridgingDelay(uint64)", _minimumBridgingDelay)
+        );
+        emit SetMinimumBridgingDelay(_minimumBridgingDelay);
+    }
+
     /**
-     * @notice Privileged account can associate a whitelisted token with its linked token address on L2 and its
-     * BridgePool address on this network. The linked L2 token can thereafter be deposited into the Deposit contract
-     * on L2 and relayed via the BridgePool contract.
+     * @notice Owner can pause/unpause L2 deposits for all tokens.
+     * @dev Only callable by the current owner. Will set the same setting in the L2 Deposit contract via the cross
+     * domain messenger.
+     * @param _depositsEnabled bool to set if the deposit box should accept/reject deposits.
+     * @param _l2Gas Gas limit to set for relayed message on L2
+     */
+    function setEnableDeposits(bool _depositsEnabled, uint32 _l2Gas) public onlyOwner depositContractSet {
+        sendCrossDomainMessage(
+            depositContract,
+            _l2Gas,
+            abi.encodeWithSignature("setEnableDeposits(bool)", _depositsEnabled)
+        );
+        emit DepositsEnabled(_depositsEnabled);
+    }
+
+    /**
+     * @notice Privileged account can associate a whitelisted token with its linked token address on L2. The linked L2
+     * token can thereafter be deposited into the Deposit contract on L2 and relayed via the BridgePool contract.
+     * @dev This will also deploy a new BridgePool if there is no such contract associated with the L1 token.
      * @dev Only callable by Owner of this contract. Also initiates a cross-chain call to the L2 Deposit contract to
      * whitelist the token mapping.
      * @param _l1Token Address of L1 token that can be used to relay L2 token deposits.
      * @param _l2Token Address of L2 token whose deposits are fulfilled by `_l1Token`.
-     * @param _bridgePool Address of pool contract that stores passive liquidity with which to fulfill deposits.
      * @param _l2Gas Gas limit to set for relayed message on L2
      */
     function whitelistToken(
         address _l1Token,
         address _l2Token,
-        address _bridgePool,
         uint32 _l2Gas
     ) public onlyOwner depositContractSet {
         require(_getCollateralWhitelist().isOnWhitelist(address(_l1Token)), "Payment token not whitelisted");
 
         L1TokenRelationships storage whitelistedToken = whitelistedTokens[_l1Token];
         whitelistedToken.l2Token = _l2Token;
+        // If token mapping isn't associated with a BridgePool, then deploy a new one. Otherwise just update the mapping.
+        if (whitelistedToken.bridgePool == address(0)) {
+            BridgePool newPool = new BridgePool(address(this), timer);
```

Comment:
this will clearly need to update once we've got token-specific info in the `BridgePool`. Should also consider what happens if you re-whitelist an existing token. What do you do with the existing `BridgePool`? is there some process that should be followed to update it?

Reply:
> this will clearly need to update once we've got token-specific info in the BridgePool.

what kind of token-specific info?

> Should also consider what happens if you re-whitelist an existing token. What do you do with the existing BridgePool? is there some process that should be followed to update it?

Yeah, hm this is a good point. For now should we just always redeploy a new pool whenever the mapping is changed?